### PR TITLE
Resolve #1215 -- Introduced a General Scripting system for each item in order to script passive effects

### LINKED ISSUE #1215: Item Passive Implementation
Passives for items are currently not implemented.

### PR DESCRIPTION
**Pull request by cabeca1143**
* Introduced `ItemPassives` namespace
* Item spells no longer use the `Spells`  namespace, they use `ItemSpells`
* "Items" folder changed, now it is organized between "Actives" and "Passives"
* Reorganized the scripts folder so now there's a "Spells", which inside you'll find the Champions and Items folder.
* Introduced `IItemScript` Interface.
* Introduced a system that assigns every item an `ItemScript`.
* Introduced Doran's Shield and Doran's Ring Passive effects as demonstration
* Updated the Tests in order to properly reflect the changes.

Resolve #1215

### CODE REVIEW

File: Content/LeagueSandbox-Scripts/Spells/Items/Passives/Doran's Shield.cs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+namespace ItemPassives
+{
+    public class ItemID_1054 : IItemScript
+    {
+        public IStatsModifier StatsModifier { get; private set; } = new StatsModifier();
+
+        public void OnActivate(IObjAiBase owner)
+        {
+            StatsModifier.HealthRegeneration.BaseBonus += 1.2f;
```
Comment (SomeOtherDev) [inline Content/LeagueSandbox-Scripts/Spells/Items/Passives/Doran's Shield.cs]: Is this correct? 
Wiki reads: `+6 health regen per 5 seconds`
https://web.archive.org/web/20150214051041/http://leagueoflegends.wikia.com/wiki/Doran's_Shield

**Review by SomeOtherDev — COMMENTED** [review]
Just one comment on Doran's Shield implementation

File: Content/LeagueSandbox-Scripts/Spells/Items/Passives/Doran's Shield.cs (hunk 1 above)
Comment (cabeca1143) [reply Content/LeagueSandbox-Scripts/Spells/Items/Passives/Doran's Shield.cs]: Yeah, 6 regen per 5 seconds translates to 1.2 per second (6/5)

**Review by danil179 — COMMENTED** [review]
I think that there are new spells from the last PRs. Please check if all the spells moved to their new locations (I also don't think that you really need to move the Champions folder into a new Spells folder. I prefer not to move many files in a PR, unless there is a justification).

**Comment by cabeca1143** [comment]
> I think that there are new spells from the last PRs. Please check if all the spells moved to their new locations (I also don't think that you really need to move the Champions folder into a new Spells folder. I prefer not to move many files in a PR, unless there is a justification).

Resolved

**Review by danil179 — APPROVED** [review]
Approved. Didn't test.

**Pull request merged**